packages/app/app/stacks/stacks-orders.tsx

```diff
@@ -0,0 +1,116 @@
+"use client";
+
+import {
+  StackOrder,
+  filterActiveOrders,
+  filterCompletedOrders,
+  getStackOrders,
+} from "@/models/stack-order";
+import { Tab } from "@headlessui/react";
+import { BodyText, ButtonLink, HeadingText } from "@/ui";
+import { StacksTable } from "@/components/StacksTable";
+import { EmptyState } from "@/app/stacks/empty-state";
+import { Order, getOrders } from "@/models/order";
+import { ChainId } from "@stackly/sdk";
+import { useEffect, useState } from "react";
+import { cx } from "class-variance-authority";
+
+export interface StackOrdersProps {
+  address: string;
+  chainId: ChainId;
+}
+
+export const StackOrders = ({ chainId, address }: StackOrdersProps) => {
+  const [loading, setLoading] = useState(true);
+  const [completedOrders, setCompletedOrders] = useState<StackOrder[]>([]);
+  const [activeOrders, setActiveOrders] = useState<StackOrder[]>([]);
+  const [currentOrders, setCurrentOrders] = useState<Order[]>([]);
+  const [currentStackOrders, setCurrentStackOrders] = useState<StackOrder[]>(
+    []
+  );
+
+  useEffect(() => {
+    async function fetchOrdersData() {
+      const orders = await getOrders(chainId, address.toLowerCase());
+      if (orders) setCurrentOrders(orders);
+    }
+    fetchOrdersData();
+  }, [address, chainId]);
+
+  useEffect(() => {
+    async function fetchStackOrdersData() {
+      const stackOrders = await getStackOrders(chainId, currentOrders);
+      if (stackOrders) setCurrentStackOrders(stackOrders);
+    }
+    fetchStackOrdersData();
+  }, [chainId, currentOrders]);
+
+  useEffect(() => {
+    if (currentStackOrders?.length) {
+      setCompletedOrders(filterCompletedOrders(currentStackOrders));
+      setActiveOrders(filterActiveOrders(currentStackOrders));
+      setLoading(false);
+    }
+  }, [currentStackOrders]);
```

Comment:
If I don't create any `currentStackOrders` it will create an infinite loading state.

Reply:
I'll fix that, I think related with bellow.